### PR TITLE
Add support for libp2p backend in integration tests

### PR DESCRIPTION
**Pull request by Zeegomo**
Ideally, we should test both version in integrations tests, I think it's possible to set up a run for each of the backends right @bacv ?

### CODE REVIEW

**Comment by bacv** [comment]
> I think it's possible to set up a run for each of the backends

Yes, integration job builds `nomos-node` and then runs `cargo test` with it. With new feature flags, these steps will be duplicated for waku and libp2p.

I can add updated jenkinsfile to this PR or create a new one, which do you prefer?

File: tests/src/nodes/nomos.rs [hunk 1]
```diff
@@ -8,12 +8,17 @@ use consensus_engine::overlay::{FlatOverlaySettings, RoundRobin};
 use consensus_engine::NodeId;
 use nomos_consensus::{CarnotInfo, CarnotSettings};
 use nomos_http::backends::axum::AxumBackendSettings;
+#[cfg(feature = "libp2p")]
+use nomos_libp2p::Multiaddr;
+use nomos_libp2p::SwarmConfig;
```
Comment (bacv) [inline tests/src/nodes/nomos.rs:13]: This probably needs to be in one line for the feature flag.
`use nomos_libp2p::{Multiaddr, SwarmConfig};`

**Review by bacv — APPROVED** [review]

**Review by youngjoon-lee — APPROVED** [review]

**Comment by danielSanchezQ** [comment]
> > I think it's possible to set up a run for each of the backends
> 
> Yes, integration job builds `nomos-node` and then runs `cargo test` with it. With new feature flags, these steps will be duplicated for waku and libp2p.
> 
> I can add updated jenkinsfile to this PR or create a new one, which do you prefer?

wouldn't the `all-features` flag work? 🤔

**Comment by bacv** [comment]
>  wouldn't the all-features flag work?

It won't work for building two separate binary files. It might work for `cargo test`, but in order to run integration tests sequentially, every test is executed with a dedicated `cargo test` command line.

**Comment by Zeegomo** [comment]
Right now we use `all-features` in tests which is making the test fail as in the binary/tests `waku`/`libp2p` are mutually exclusive features. Can we merge https://github.com/logos-co/nomos-node/pull/290 first?

**Pull request closed without merging**